stopify-continuations/src/runtime/abstractRuntime.ts

```diff
@@ -171,15 +170,15 @@ export abstract class Runtime {
         }
         else if(result.type === 'normal') {
           assert(this.mode, 'execution completed in restore mode');
-          return result.value;
+          return onDone();
         }
         else if(result.type === 'exception') {
           assert(this.mode,
             `execution completed in restore mode, error was: ${result.value}`);
-          throw result.value;
```

Comment:
I don't believe there was any point in throwing or returning these values. Since the runtime system usually terminates on another turn (i.e., after yielding), the exception and return value would be lost. We could send the exception / result to the new onDone callback that I have defined.